web/assets/js/main_out.js

```diff
@@ -1325,239 +1360,252 @@
                 curRl = (9 * curRl + this.s) / 10;
                 curP.rl = (prevRl + nextRl + 8 * curRl) / 10;
 
-                var angle = 2 * Math.PI * i / len;
-                var rl = curP.rl;
+                const angle = 2 * Math.PI * i / this.points.length;
+                let rl = curP.rl;
                 if (this.jagged && i % 2 == 0) {
                     rl += 5;
                 }
                 curP.x = this.x + Math.cos(angle) * rl;
                 curP.y = this.y + Math.sin(angle) * rl;
             }
-        },
-        parseName: function(value) { // static method
-            value = value || "";
-            var nameAndSkin = /^(?:\{([^}]*)\})?([^]*)/.exec(value);
-            return {
-                name: nameAndSkin[2].trim(),
-                skin: (nameAndSkin[1] || "").trim() || nameAndSkin[2]
-            };
-        },
-        setName: function(name) {
-            var nameAndSkin = Cell.prototype.parseName(name);
+        }
+        setName(name) {
+            const nameAndSkin = Cell.parseName(name);
             this.name = nameAndSkin.name;
             this.setSkin(nameAndSkin.skin);
-        },
-        setSkin: function(value) {
-            this.skin = (value && value[0] === "%" ? value.slice(1) : value) || this.skin;
-            if (this.skin === null || !knownSkins.hasOwnProperty(this.skin) || loadedSkins[this.skin])
+        }
+        setSkin(value) {
+            this.skin = (value && value[0] === '%' ? value.slice(1) : value) || this.skin;
+            if (this.skin === null || !knownSkins.has(this.skin) || loadedSkins.has(this.skin)) {
                 return;
-            loadedSkins[this.skin] = new Image();
-            loadedSkins[this.skin].src = SKIN_URL + this.skin + ".png";
-        },
-        setColor: function(value) {
-            if (!value) { log.warn("got no color"); return; }
+            }
+            const skin = new Image();
+            skin.src = `${SKIN_URL}${this.skin}.png`;
+            loadedSkins.set(this.skin, skin);
+        }
+        setColor(value) {
+            if (!value) {
+                Logger.warn('Got no color');
+                return;
+            }
             this.color = value;
-            this.sColor = darkenColor(value);
-        },
-        draw: function(ctx) {
+            this.sColor = value.darker();
+        }
+        draw(ctx) {
             ctx.save();
             this.drawShape(ctx);
             this.drawText(ctx);
             ctx.restore();
-        },
-        drawShape: function(ctx) {
-            ctx.fillStyle = settings.showColor ? this.color : Cell.prototype.color;
-            ctx.strokeStyle = settings.showColor ? this.sColor : Cell.prototype.sColor;
+        }
+        drawShape(ctx) {
+            ctx.fillStyle = settings.showColor ? this.color.toHex() : Cell.prototype.color;
+            ctx.strokeStyle = settings.showColor ? this.sColor.toHex() : Cell.prototype.sColor;
```

Comment:
You have a bug here, `Cell.prototype.color` and `Cell.prototype.sColor` don't exist.